src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -503,11 +518,133 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
                     "model and its config, please check passed config.");
 
     implement_properties();
+
     LOG_DEBUG("Done");
 }
 
-void ov::npuw::LLMCompiledModel::export_model(std::ostream& model) const {
-    OPENVINO_NOT_IMPLEMENTED;
+void ov::npuw::LLMCompiledModel::export_model(std::ostream& stream) const {
+    LOG_INFO("Serializing LLMCompiledModel...");
+    LOG_BLOCK();
+
+    using namespace ov::npuw::s11n;
+
+    // Serialize magic number first
+    write(stream, NPUW_SERIALIZATION_INDICATOR);
+
+    // Serialize general meta info
+    write(stream, OPENVINO_VERSION_MAJOR);
+    write(stream, OPENVINO_VERSION_MINOR);
+    write(stream, OPENVINO_VERSION_PATCH);
+
+    // Serialize name first + NPUW identifier for deserialization
+    // FIXME: consider a better approach then using name there
+    std::string name = std::string(NPUW_NAME_IDENTIFIER) + m_name;
+    write(stream, name);
+
+    // Serialize inputs and outputs
+    write(stream, inputs());
+    write(stream, outputs());
+
+    // Serialize LLMCompiledModel-specific data
+    write(stream, m_kvcache_desc.max_prompt_size);
+    write(stream, m_kvcache_desc.total_size);
+    write(stream, m_kvcache_desc.num_stored_tokens);
+    write(stream, m_kvcache_desc.dim);
+
+    // Serialize CompiledModels
+    m_kvcache_compiled->serialize(stream);
+    m_prefill_compiled->serialize(stream);
+
+    // Serialize configs
+    write(stream, m_kvcache_compiled->m_cfg);
+    write(stream, m_prefill_compiled->m_cfg);
+
+    // Serialize weights bank (if required)
+    const auto& kv_bank = m_kvcache_compiled->m_weights_bank;
+    const auto& p_bank = m_prefill_compiled->m_weights_bank;
+    NPUW_ASSERT(kv_bank && p_bank && kv_bank == p_bank && "Prefill and KVCache models' weight bank should be shared!");
+    // FIXME: support weightless flow
+    write(stream, kv_bank->get_name());
+    kv_bank->serialize(stream);
+
+    LOG_INFO("Done.");
+}
+
+std::shared_ptr<ov::npuw::LLMCompiledModel> ov::npuw::LLMCompiledModel::deserialize(
+    std::istream& stream,
+    const std::shared_ptr<const ov::IPlugin>& plugin,
+    const ov::AnyMap& properties) {
+    LOG_INFO("Deserializing LLMCompiledModel...");
+    LOG_BLOCK();
+
+    using namespace ov::npuw::s11n;
+
+    // Sanity check magic number
+    uint64_t serialization_indicator = 0;
+    read(stream, serialization_indicator);
+    NPUW_ASSERT(serialization_indicator == NPUW_SERIALIZATION_INDICATOR && "This blob wasn't serialized via NPUW!");
+
+    // Deserialize general meta info
+    int vmajor, vminor, vpatch;
+    read(stream, vmajor);
+    read(stream, vminor);
+    read(stream, vpatch);
+
+    NPUW_ASSERT(vmajor == OPENVINO_VERSION_MAJOR && vminor == OPENVINO_VERSION_MINOR &&
+                vpatch == OPENVINO_VERSION_PATCH && "Only blobs serialized with the same OV version are supported!");
```

Comment:
I'd rather throw an exception here & include the deserialized version number vs the expected one in the error text. That'd be more helpful for the end user